jerry-core/parser/js/js-parser-statm.c

```diff
@@ -1350,15 +1350,26 @@ parser_parse_for_statement_start (parser_context_t *context_p) /**< context */
           parser_emit_cbc_ext (context_p, is_for_in ? CBC_EXT_FOR_IN_GET_NEXT
                                                     : CBC_EXT_FOR_OF_GET_NEXT);
 
+          parser_pattern_flags_t flags = (PARSER_PATTERN_BINDING | PARSER_PATTERN_TARGET_ON_STACK);
+
           if (context_p->next_scanner_info_p->source_p == (context_p->source_p + 1))
           {
-            JERRY_ASSERT (context_p->next_scanner_info_p->type == SCANNER_TYPE_INITIALIZER);
+            if (context_p->next_scanner_info_p->type == SCANNER_TYPE_INITIALIZER)
```

Comment:
Would it be worth to drop SCANNER_TYPE_INITIALIZER in favor of SCANNER_TYPE_LITERAL_FLAGS?

Reply:
That I don't know, I've just did a quick "fix" based on the existing codes. At the moment I did not went deeper on the type of scanners and how the `SCANNER_TYPE_INITIALIZER` used/works.

Reply:
@zherczeg If we don't need the initializer one I would remove it in another PR.

Reply:
Please don't forget about it.